system-addon/content-src/components/Card/Card.jsx

```diff
@@ -27,6 +29,10 @@ class Card extends React.Component {
       showContextMenu: true
     });
   }
+  onClick(event) {
```

Comment:
yeah I agree, that would be clearer. Also can you add a comment somewhere around here about why this exists? (i.e. that we need to override the regular behaviour because of HTTP referrer stuff)